distributed/scheduler.py

```diff
@@ -5379,16 +5379,27 @@ def adaptive_target(self, comm=None, target_duration=None):
             cpu = max(1, cpu)
 
         # Memory
-        limit_bytes = {addr: ws.memory_limit for addr, ws in self.workers.items()}
+        limit_bytes = [ws.memory_limit for ws in self.workers.values()]
         worker_bytes = [ws.nbytes for ws in self.workers.values()]
-        limit = sum(limit_bytes.values())
+        limit = sum(limit_bytes)
         total = sum(worker_bytes)
-        if total > 0.6 * limit:
+        if total < 0.6 * limit or limit <= 0:
             memory = 2 * len(self.workers)
         else:
             memory = 0
 
         target = max(memory, cpu)
+
+        # runnable tasks
+        tasks = len(
+            [
+                task
+                for task in self.tasks.values()
+                if task.state in ("processing", "no-worker") and not task.waiting_on
```

Comment:
do we also need `released` and `memory` here? I don't think we need `erred` and `forgotten`.

Thank you for having a nice overview of the states here :) https://distributed.dask.org/en/latest/scheduling-state.html

Reply:
does iterating over all tasks take potentially too much time? We could stop as soon as we have more than `target` tasks.